### PR TITLE
docs(RSP): add TreeView docs

### PR DESCRIPTION
**Pull request by reidbarber**
## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

## 📝 Test Instructions:

Check new docs page: https://reactspectrum.blob.core.windows.net/reactspectrum/7a245fb905688e49c6182ee4ba4683192bf332f0/docs/react-spectrum/TreeView.html

### CODE REVIEW

File: packages/@react-spectrum/tree/docs/TreeView.mdx [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+```
+
+## Content
+TreeView is a [collection component](../react-stately/collections.html) that provides users with a way to navigate nested hierarchical information.
+
+Basic usage of TreeView, seen in the example above, shows the use of a static collection where the contents of the TreeView are hard coded. Dynamic collections, as shown below, can be used when the options come from an external data source, such as an API, or update over time. Providing the data in this way allows TreeView to automatically cache the rendering of each item, which dramatically improves performance.
+
+Items can be statically defined as children, or generated dynamically using a function based on the data passed to the `items` prop.
+
+Each item has a unique key defined by the data. The `key` of each item element is implicitly defined by the id property of the item object. See [collections](../react-stately/collections.html#unique-keys) to learn more keys in dynamic collections.
```
Comment (snowystinger) [inline packages/@react-spectrum/tree/docs/TreeView.mdx:97]: while relevant, this technically points to the wrong collection implementation. We should point this to https://react-spectrum.adobe.com/react-aria/collections.html#unique-ids instead

File: packages/@react-spectrum/tree/docs/TreeView.mdx [hunk 2]
```diff
@@ -0,0 +1,512 @@
...
+    <TreeViewItem id="image-3" textValue="Image 3">
+      <Text>Image 3</Text>
+      <Image />
+    </TreeViewItem>
+  </TreeViewItem>
+</TreeView>
+```
+
+## Content
+TreeView is a [collection component](../react-stately/collections.html) that provides users with a way to navigate nested hierarchical information.
```
Comment (snowystinger) [inline packages/@react-spectrum/tree/docs/TreeView.mdx:91]: same thing here, this should point to react-aria/collections

File: packages/@react-spectrum/tree/docs/TreeView.mdx [hunk 3]
```diff
@@ -0,0 +1,512 @@
...
+To internationalize a TreeView, all text content within the TreeView should be localized. This includes the `aria-label` provided to the TreeView if any.
+For languages that are read right-to-left (e.g. Hebrew and Arabic), the layout of TreeView is automatically flipped.
+
+## Labeling
+### Accessibility
+An `aria-label` must be provided to the TreeView for accessibility. If the TreeView is labeled by a separate element, an `aria-labelledby` prop must be provided using the id of the labeling element instead.
+
+## Expansion
+
+By default, TreeView items are initially collapsed. Use `defaultExpandedKeys` to provide a default set of expanded items. Note that the value of the expended keys must match the `id` prop of the TreeViewItem.
```
Comment (snowystinger) [inline packages/@react-spectrum/tree/docs/TreeView.mdx:141]:
```suggestion
By default, TreeView items are initially collapsed. Use `defaultExpandedKeys` to provide a default set of expanded items. Note that the value of the expanded keys must match the `id` prop of the TreeViewItem.
```

File: packages/@react-spectrum/tree/docs/TreeView.mdx [hunk 4]
```diff
@@ -0,0 +1,512 @@
...
+and the selected items are displayed with a highlighted background instead.
+
+In addition to changing the appearance, the selection behavior also changes depending on the `selectionStyle` prop. In the default checkbox selection style, clicking, tapping, or pressing
+the <Keyboard>Space</Keyboard> or <Keyboard>Enter</Keyboard> keys toggles selection for the focused item. Using the arrow keys moves focus but does not change selection.
+
+In the highlight selection style, however, clicking a item with the mouse _replaces_ the selection with only that item. Using the arrow keys moves both focus and selection. To select multiple items,
+modifier keys such as <Keyboard>Ctrl</Keyboard>, <Keyboard>Cmd</Keyboard>, and <Keyboard>Shift</Keyboard> can be used. On touch screen devices, selection always behaves as toggle since modifier
+keys may not be available.
+
+These selection styles implement the behaviors defined in [Aria Practices](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/#keyboardinteraction).
```
Comment (snowystinger) [inline packages/@react-spectrum/tree/docs/TreeView.mdx:411]: listbox? or do we implement https://www.w3.org/WAI/ARIA/apg/patterns/treeview/ ?

**Review by snowystinger — COMMENTED** [review]
Screenshot 2024-08-06 at 10 29 12 AM
I don't think this is from the docs, but the vertical centering on some items isn't quite right. Seems to be the folders?
@LFDanLu if you've noticed this?

It'd be good to get one of these treeviews into a container that forces scrolling. Maybe generate a larger data set and hide it like we do for CSS?

File: packages/@react-spectrum/tree/docs/TreeView.mdx (hunk 1 above)
Comment (reidbarber) [reply packages/@react-spectrum/tree/docs/TreeView.mdx]: Thanks for catching these!

**Comment by LFDanLu** [comment]
@snowystinger @reidbarber I think the weird centering is happening because the rows aren't the right height, they should be 40px tall, but they are only 35px here (probably cuz the font size is 14px on the page)

**Review by snowystinger — APPROVED** [review]

File: packages/@react-spectrum/tree/docs/TreeView.mdx [hunk 5]
```diff
@@ -0,0 +1,512 @@
...
+```
+
+## Content
+TreeView is a [collection component](../react-aria/collections.html) that provides users with a way to navigate nested hierarchical information.
+
+Basic usage of TreeView, seen in the example above, shows the use of a static collection where the contents of the TreeView are hard coded. Dynamic collections, as shown below, can be used when the options come from an external data source, such as an API, or update over time. Providing the data in this way allows TreeView to automatically cache the rendering of each item, which dramatically improves performance.
+
+Items can be statically defined as children, or generated dynamically using a function based on the data passed to the `items` prop.
+
+Each item has a unique key defined by the data. The `key` of each item element is implicitly defined by the id property of the item object. See [collections](../react-aria/collections.html#unique-ids) to learn more keys in dynamic collections.
```
Comment (yihuiliao) [inline packages/@react-spectrum/tree/docs/TreeView.mdx:97]:
```suggestion
Each item has a unique key defined by the data. The `key` of each item element is implicitly defined by the id property of the item object. See [collections](../react-aria/collections.html#unique-ids) to learn more about keys in dynamic collections.
```

**Review by yihuiliao — COMMENTED** [review]
Do we want to add examples of TreeView with ActionMenu or ActionGroup?

**Comment by reidbarber** [comment]
@yihuiliao Good idea, just added

**Review by yihuiliao — APPROVED** [review]

File: packages/@react-spectrum/tree/docs/TreeView.mdx [hunk 6]
```diff
@@ -0,0 +1,604 @@
...
+}
+
+<TreeView aria-label="Example tree for empty state" height="size-2400" renderEmptyState={renderEmptyState}>
+  {[]}
+</TreeView>
+```
+
+## Testing
+
+The TreeView features automatic virtualization and may need specific mocks in a test environment to enable said virtualization properly.
```
Comment (LFDanLu) [inline packages/@react-spectrum/tree/docs/TreeView.mdx:590]: The TreeView isn't actually virtualized so this sentence will need to be removed along with the link below. Thanks for including this section though :)

**Review by LFDanLu — CHANGES_REQUESTED** [review]
Looks good! Just a couple of comments and one change

**Review by LFDanLu — DISMISSED** [review]
LGTM

File: packages/@react-spectrum/tree/docs/TreeView.mdx [hunk 7]
```diff
@@ -397,7 +267,14 @@ You can disable specific items by providing an array of keys to TreeView via the
 
 ```tsx example
 // Using the same tree as above
```
Comment (yihuiliao) [inline packages/@react-spectrum/tree/docs/TreeView.mdx:269]: mega nit: i don't think we need to have the comment here anymore given that a lot of these examples use `<ExampleTree>` now. same goes for the examples below as well

**Review by yihuiliao — COMMENTED** [review]
looks good! really just a small thing that i commented on

**Review by yihuiliao — DISMISSED** [review]

File: packages/@react-spectrum/tree/docs/TreeView.mdx [hunk 8]
```diff
@@ -0,0 +1,498 @@
...
+## Content
+TreeView is a [collection component](../react-aria/collections.html) that provides users with a way to navigate nested hierarchical information.
+
+Basic usage of TreeView, seen in the example above, shows the use of a static collection where the contents of the TreeView are hard coded. Dynamic collections, as shown below, can be used when the options come from an external data source, such as an API, or update over time. Providing the data in this way allows TreeView to automatically cache the rendering of each item, which dramatically improves performance.
+
+Items can be statically defined as children, or generated dynamically using a function based on the data passed to the `items` prop.
+
+Each item has a unique key defined by the data. The `key` of each item element is implicitly defined by the id property of the item object. See [collections](../react-aria/collections.html#unique-ids) to learn more about keys in dynamic collections.
+
+<strong>Note: Asyncronous tree loading (i.e, loading by level or infinite scrolling) is not yet supported in TreeView.</strong>
```
Comment (snowystinger) [inline packages/@react-spectrum/tree/docs/TreeView.mdx:103]:
```suggestion
<strong>Note: Asynchronous tree loading (i.e, loading by level or infinite scrolling) is not yet supported in TreeView.</strong>
```

**Review by snowystinger — APPROVED** [review]

**Review by yihuiliao — APPROVED** [review]

**Pull request merged**